feat(hesai): agnocast support

mojomex opened the pull request:
## PR Type

- New Feature

## Related Links

<!-- Please write related links to GitHub/Jira/Slack/etc. -->

## Description

Agnocast support for Hesai pointcloud output.

This PR does the following:
* introduce an adapted version of [autoware_agnocast_wrapper.hpp](https://github.com/autowarefoundation/autoware_universe/blob/3b8bdb1d9c2e6f235d1c1d3773198ba7e8351d34/common/autoware_agnocast_wrapper/include/autoware/agnocast_wrapper/autoware_agnocast_wrapper.hpp)
* adds `agnocastlib` as an optional dependency
* indroduces the wrapper macros in all places where Hesai pointclouds, packets or images (e.g. blockage mask) are published/handled

## Self-Evaluation

With our bus product (experimental branch) [TIER IV INTERNAL](https://github.com/tier4/pilot-auto.x2/tree/experiment/v4.3.0).
Autoware Universe version used: https://github.com/autowarefoundation/autoware_universe/pull/10812
AIP Launcher version used: https://github.com/tier4/aip_launcher/tree/experiment/v4.3.0

With this rosbag: [TIER IV INTERNAL](https://console.mob.tier4.jp/projects/x2_dev/rosbag?viz=stream&agg=count&from_ts=1747722780&to_ts=1750314780&live=true&query=log_file_name%3A%28cfa23601-97c4-4d47-a37e-edfc7e080d8c_2025-05-29-17-00-12_p0900_11.db3%29&file_id=e4c77db5-5700-4442-9317-9be5e85a77ca)

Performance analysis: [TIER IV INTERNAL](https://star4.slack.com/archives/C057JDFS5M2/p1750257474104259?thread_ts=1749812789.415529&cid=C057JDFS5M2)

Demo video:


https://github.com/user-attachments/assets/7c75b952-48db-4f93-b005-9d920818faca

## Review Procedure

Putting this PR into the above product version will result in build errors in various modules, but sensing is unaffected.
Just make sure `autoware_launch` is built and launch with only sensing enabled.

<!-- Explain how to review this PR. -->

## Remarks

<!-- Write remarks as you like if you need them. -->

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR**: Required to pass before the merge.

atsushi421 commented on build_depends.repos:
```diff
@@ -14,3 +14,7 @@ repositories:
     type: git
     url: https://github.com/autowarefoundation/autoware_msgs
     version: main
+  agnocast:
```
```suggestion
  # TODO(TIER IV): During the transition period of Agnocast introduction,
  # the Agnocast ROS packages are provided as a source build.
  # Once the transition stabilizes, use the packages released from the official ROS repository.
  # Issue: https://github.com/autowarefoundation/autoware/issues/5968
  agnocast:
```

mojomex replied:
Thank you, applied in ad8096d.

mojomex commented:
@drwnz Thank you for the review! I have since
* added a guide on how to use with Agnocast in the `README.md`
* removed Agnocast support for the `/pandar_packets` topic as discussed internally

drwnz reviewed (approved):
LGTM!

@atsushi421 @sykwer 
Currently, if built with agnocast support the build will succeed even if there is an issue with the kernel module, and there will be a run-time error. Would there be benefit to check that the kernel module is loaded at build time with `lsmod` via `execute_process` in CMake? I believe Cmake files testing for CUDA availability for GPU/CPU conditional building often use this method.

atsushi421 commented:
@drwnz 
Hi David-san, Thank you for your thoughtful comments.

We [discussed this within the team](https://star4.slack.com/archives/C07FL8616EM/p1750892539391739) and have decided not to address it at this time :pray:
The reason is that our current approach is to have `agnocast.ko` automatically loaded at boot via the [configuration file](https://github.com/tier4/pilot-auto.x2/blob/beta/v4.2/ansible/playbooks/roles/agnocast/tasks/main.yaml#L75-L82). Therefore, it is not necessary to guarantee that it is inserted at build time, and doing so could potentially inconvenience users.

We appreciate your continued support.
